gameplay/src/Transform.cpp

```diff
@@ -99,43 +99,27 @@ const char* Transform::getTypeName() const
 
 const Matrix& Transform::getMatrix() const
 {
-    if (_matrixDirtyBits)
+    if (_matrixDirtyBits & (DIRTY_TRANSLATION | DIRTY_ROTATION | DIRTY_SCALE))
     {
         if (!isStatic())
         {
-            bool hasTranslation = !_translation.isZero();
             bool hasScale = !_scale.isOne();
             bool hasRotation = !_rotation.isIdentity();
 
             // Compose the matrix in TRS order since we use column-major matrices with column vectors and
             // multiply M*v (as opposed to XNA and DirectX that use row-major matrices with row vectors and multiply v*M).
-            if (hasTranslation || (_matrixDirtyBits & DIRTY_TRANSLATION) == DIRTY_TRANSLATION)
```

Comment:
The code is quite convoluted, for no real gain. Almost all real-life matrices actually have translation. And the additional branches are hard on CPU instruction scheduling. The real FPU cost comes in the rotation case, which makes that branch worth it. The rest of the code really just ends up paying for filling the cache. Thus, in practice, this code is simpler, and runs faster, on most architectures (including ARM 8.) But I'm mostly after the simplicity :-)